File: SiteVisitsNetCore2020Web/ClientApp/src/app/viewmodels/VisitViewModel.ts
import { Visit } from "../models/Visit";

export class VisitViewModel {
  visit: Visit;
  ipAddress: string;
  combinedTerms: string;
  location: string;
  pageUrl: string;
  pageTitle: string;
  cameFrom: string;

  constructor(options: {
    visit: any,
    ipAddress: string,
    combinedTerms: string,
    location: string,
    pageUrl: string,
    pageTitle: string,
    cameFrom: string
  }) {
    this.cameFrom = options.cameFrom;
    this.combinedTerms = options.combinedTerms;
    this.ipAddress = options.ipAddress;
    this.location = options.location;
    this.pageTitle = options.pageTitle;
    this.pageUrl = options.pageUrl;
    this.visit = new Visit(options.visit);
  }

}

File: SiteVisitsNetCore2020Web/ViewModels/VisitViewModel.cs
using SiteVisitsNetCore2020Web.Models;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Threading.Tasks;

namespace SiteVisitsNetCore2020Web.ViewModels
{
    public class VisitViewModel
    {
        public Visit Visit { get; set; }
        public string IpAddress { get => Visit.IpAddress?.IpV4Address; }
        public string CombinedTerms { get; set; }
        public string Location { get; set; }
        public string PageUrl { get; set; }
        public string PageTitle { get; set; }
        public string CameFrom { get; set; }
    }
}
